### PR TITLE
Add pyxem and lumispy extensions and all `signal_type`.

### PR DESCRIPTION
**Pull request by ericpre**
As mentioned, https://github.com/hyperspy/hyperspy/pull/2434#discussion_r445639663, add `signal_type` to make them easier to find.

- [x] Update extension list,
- [x] add all `signal_type`.

### CODE REVIEW

**Comment by francisco-dlp** [comment]
I have some reserves about this:

1. I think that it is up to the packages developers to list their package here, so we should get their opinion before merging this
2. Adding the signal subclasses provided by the different packages is great. However, it'll be hard to keep the table up-to-date manually. In a way, obsolete information may be worst that no information on this. An alternative would be to automatically generate the list.

**Comment by ericpre** [comment]
> I have some reservations about this:
> 
> 1. I think that it is up to the packages developers to list their package here, so we should get their opinion before merging this

In principle, it would be better indeed, but more practically, this is not always happen (easy to forgot/not to pay attention), even if this is documented https://hyperspy.readthedocs.io/en/latest/dev_guide/writing_extensions.html. For me, it is more of community thing (documentation, support users), so anyone should can do it.

> 2. Adding the signal subclasses provided by the different packages is great. However, it'll be hard to keep the table up-to-date manually. In a way, obsolete information may be worst that no information on this. An alternative would be to automatically generate the list.

Yes, I was thinking about this, we need a script which generate the table for the `README.md` file and also add the components. I will update the PR.
Longer term, we should also think of a similar way to document signal metadata.

**Comment by ericpre** [comment]
What about inviting the extension developer to the team of this repository, so that we can use this repository as communication channel between/with extension developers?

**Comment by francisco-dlp** [comment]
I agree that we it'll be a good idea to implement such a communication channel. However, wouldn't a gitter room be more suitable?

**Comment by ericpre** [comment]
gitter is good for quick chat or to help each other but awful for discussion or to find information, so I am not convinced it would be useful in this case.

**Comment by francisco-dlp** [comment]
@ericpre, which are your current thoughts on how to implement this?

Some ideas that come to my mind in case that they can be useful:

* The list of extensions could be kept in json format
* The ``hs.print_known_signal_types()`` function could optionally use the functionality implemented here to print the known signal types of packages that are not installed locally. Therefore, it would be good to have a clean API for external consumption.

**Comment by dnjohnstone** [comment]
> 1. I think that it is up to the packages developers to list their package here, so we should get their opinion before merging this

Thanks @ericpre for doing this. I think I can probably sign off on both pyxem and lumispy - it was on my list to do this and just didn't get there yet, so much appreciated yet. Will quickly review now.

> 2. Adding the signal subclasses provided by the different packages is great. However, it'll be hard to keep the table up-to-date manually. In a way, obsolete information may be worst that no information on this. An alternative would be to automatically generate the list.

Automatically generating the list would be great. Perhaps it's not so difficult to scrape the hyperspy_extension.yaml files in exentsion packages as part of the release cycle?

File: README.md [hunk 1]
```diff
@@ -2,10 +2,17 @@
...
+| [hyperspy-gui-ipywidgets](https://github.com/hyperspy/hyperspy_gui_ipywidgets) | ipywidgets widgets for HyperSpy                                      |                                                                  |
+| [hyperspy-gui-traitsui](https://github.com/hyperspy/hyperspy_gui_traitsui)     | traitsui widgets for HyperSpy                                        |                                                                  |
+| [KikuchiPy](https://github.com/kikuchipy/kikuchipy)                            | Processing and analysis of electron backscatter diffraction patterns | `EBSD`, `EBSDMasterPattern`, `VirtualBSEImage`                   |
+| [pyXem](https://github.com/pyxem/pyxem)                                        | Multi-dimensional diffraction microscopy                             | `diffraction`, `electron_diffraction`, `diffraction_vectors`, `template_matching`, `vector_matching`, `crystallographic_map`, `vdf_image`, `reduced_intensity`, `diffraction_variance`, `dpc`, `correlation`, `power` |
```
Comment (dnjohnstone) [inline README.md:13]: Can we keep lowercase only please i.e. pyxem?

I think the list of signal types for the latest release version should be:

`diffraction`, `electron_diffraction`,  `polar_diffraction`, `diffraction_vectors`, `template_matching`, `vector_matching`, `crystallographic_map`,  `tensor_field`,  `vdf_image`,  `reduced_intensity`, `pair_distribution_function`, `diffraction_variance`, `correlation`, `power`

The dpc signals will only be introduced in pyxem-0.12.0 which is not yet released.

**Comment by ericpre** [comment]
@dnjohnstone, I suggest to continue this PR in #3.

**Comment by ericpre** [comment]
Closing in favour of #3.

**Pull request closed without merging**